newrelic-agent/src/main/java/com/newrelic/agent/config/BaseConfig.java

```diff
@@ -141,6 +136,9 @@ static Object parseValue(String val) {
         if (override != null) {
             return castValue(key, override, defaultVal);
         }
+        if (propVal instanceof ObscuredYamlPropertyWrapper) {
+            throw new IllegalArgumentException("Property " + key + " was obscured; obscured values are not supported for this property.");
```

Comment:
looking through the code it seems like this should never get hit. Can we add a comment that says that or if not true, a comment that explains the case when this would get hit?

Reply:
Here is the situation where it gets hit.   

If the user does not provide a obscuring_key by yaml , sys, or env prop,  the obscuring_key will be null.  When a yaml obscured config property is accessed during agent start up, this logic will get hit. 

Since the property is still a `ObscuredYamlPropertyWrapper`, the exception will be thrown. 

```
2020-09-23T14:40:44,539-0700 [24688 1] com.newrelic INFO: New Relic Agent: Loading configuration file "/Users/xxia/dev/integrate/newrelic/./newrelic.yml"
2020-09-23T14:40:44,685-0700 [24688 1] com.newrelic ERROR: Unable to start the New Relic Agent. Your application will continue to run but it will not be monitored.
java.lang.IllegalArgumentException: Property license_key was obscured; obscured values are not supported for this property.
	at com.newrelic.agent.config.BaseConfig.getProperty(BaseConfig.java:140) ~[newrelic.jar:6.1.0-SNAPSHOT]
```

Reply:
If the only case that this will get hit is if we don't have an obfuscation key and we know we don't have it earlier than right here then I think we should fail earlier than this since we can't run with obfuscated values. I would throw an exception here: https://github.com/newrelic/newrelic-java-agent/pull/68/files#diff-52af0964ba9cd5f4f48951c181d22098R43.

Reply:
Ok, moved it